interp/notation.ml

```diff
@@ -1992,17 +2153,17 @@ let interp_notation_as_global_reference ?loc ~head test ntn sc =
   let ntns = browse_notation true ntn scopes in
   let refs = List.map (global_reference_of_notation ~head test) ntns in
   match Option.List.flatten refs with
-  | [_,_,ref] -> ref
+  | [Some true,_ (* why not if the only one? *),_,_,ref] -> ref
```

Comment:
I don't understand this comment. Is it the same as saying that the following should be correct?
```suggestion
  | [on_parsing,_,_,_,ref] -> assert (on_parsing = Some true); ref
```

Reply:
It is not the same. If the notation is only printing, we may not want a notation string, as in say `Print "#"`, to be interpreted as a global reference.

I'm not sure this line is actually not redundant with what follows and this is not directly to this PR, so I removed the comment (sorry).